lib/output_artifact.py

```diff
@@ -25,15 +27,22 @@ class MissingOutput(Exception):
 
 @dataclasses.dataclass
 class Copier:
-    """Copy output artifacts to a directory, raising if they don't exist"""
+    """Copy output artifacts to a directory, raising MissingOutput if they don't exist"""
 
     artifacts_dir: pathlib.Path
     job_args: dict
 
 
     def copy_output_file(self, fyle):
         try:
-            shutil.copy(fyle, self.artifacts_dir)
+            if os.path.isfile(fyle):
+                shutil.copy(fyle, self.artifacts_dir)
+                return
+            if os.path.isdir(fyle):
+                shutil.copytree(fyle, self.artifacts_dir, dirs_exist_ok=True)
+                return
+            logging.warning("Can't copy %s: expected a file or a directory", fyle)
```

Comment:
I think this will cause a lot of logging if we are using phony outputs. Should this be moved to after line 47?

Reply:
I just deleted the warning.  This silence is no worse than the original silence. The scenario motivating the warning involved no phony outputs, so weaving this into the exception handler would mess it up too much.